--- ghostscript-5.50.spec.orig	Fri Apr 28 09:40:44 2000
+++ ghostscript-5.50.spec	Fri Apr 28 10:15:28 2000
@@ -3,6 +3,7 @@
 %define version	5.50
 %define hpdjver 2.6
 %define md2kver 0.2a
+%define gpver 3.1.3
 Version: %{version}
 Release: 1
 License: GPL
@@ -15,6 +16,7 @@
 Source4: ftp://ftp.cs.wisc.edu/pub/ghost/aladdin/gs550/ghostscript-%{version}gnu.tar.gz
 Source5: http://plaza26.mbn.or.jp/~higamasa/gdevmd2k/gdevmd2k-%{md2kver}.tar.gz
 Source6: http://lcewww.et.tudelft.nl/~haver/cgi-bin/download/linux/epson740.tgz
+Source7: http://download.sourceforge.net/gimp-print/print-3.1.3.tar.gz
 Patch0: ghostscript-5.50-config.patch
 Patch1: gs5.50-rth.patch
 Patch3: gs5.50-hp8xx.patch
@@ -75,6 +77,17 @@
 
 %patch6 -p1 -b .lx7000
 
+# Add gimp-print (stp) driver
+cd tmp
+tar xzvf %{SOURCE7}
+cd print-%{gpver}/Ghost
+cp gdevstp-escp2.c gdevstp-print.h gdevstp-util.c gdevstp-dither.c \
+   gdevstp-printers.c gdevstp.c gdevstp.h ../../..
+cat contrib.mak.addon >>../../../contrib.mak
+cp -p README ../../../README.stp
+cd ../../..
+perl -pi -e 's/^DEVICE_DEVS6=/DEVICE_DEVS6=\$(DD)stp.dev /g' unix-gcc.mak
+
 rm -rf tmp
 ln -s unix-gcc.mak Makefile
 
@@ -99,6 +112,7 @@
 ln -sf gs.1.gz $RPM_BUILD_ROOT/usr/man/man1/ghostscript.1.gz
 ln -sf gs $RPM_BUILD_ROOT/usr/bin/ghostscript
 strip -R .comment $RPM_BUILD_ROOT/usr/bin/gs
+cp -p README.stp /usr/doc/ghostscript-%{PACKAGE_VERSION}
 
 %clean
 rm -rf $RPM_BUILD_ROOT
